test/README.md

```diff
@@ -0,0 +1,74 @@
+
+# 1. Overview
+
+This repo add RPMI tests for rpmi framework implemented on quemu
+
+# 2. Running tests
+
+## 2.1 Running qemu
+
+Execute qemu instance with shared memory backend for RAM:
```

Comment:
These tests are mock test which we can directly run as application on x86 host so no need to launch QEMU.

Reply:
can be yes, but some has to write the server if they're not using the qemu right ?

Reply:
better to have both client and server in the same test

Reply:
Basically each test should define the messages exchanged client and server. The test fails if expected response is not received a particular message.

Reply:
One test can be something like this:

1. Enqueue a request (message)
2. Process the context to get response
3. Dequeue the response (message)
4. Inspect the response to declare pass/fail

Reply:
If possible, a summary of all tests passed/failed too

Reply:
These are already done:
- Basically each test should define the messages exchanged client and server. The test fails if expected response is not received a particular message.

- a summary of all tests passed/failed too, in the end , please refer to README.md inside test folder for sample output

Reply:
> better to have both client and server in the same test

Server code inside librpmi would bloat it, as we'll have to paste all of the qemu RPMI implentation in the server code. better to point them to upstream qemu branch which already has all of it and provide samepl qemu command , like we have in the README file no ?

Reply:
> One test can be something like this:
> 
> 1. Enqueue a request (message)
> 2. Process the context to get response
> 3. Dequeue the response (message)
> 4. Inspect the response to declare pass/fail

This is done in _base* file, please check

Reply:
Okay, sounds good.

But, we still don't need to launch QEMU for running librpmi test applications.

Reply:
We do, for now base group might work with small snippet to initialize that, but for other groups like HSM, CPPC and all still needs to form the response another server app still needs to emulate, which essentially is being done in qemu.

If we dont want to do that, then, we'll have to do one of these:
1. Use the running qemu instance
2. write a test server app inside qemu code which links all the rpmi code
3. write a test server app out side qemu copy all the rpmi code.
So I still think option#1 above seems simplest one from users point of view.

Let me know if you guys would like to implement any one of teh above still or any other server emulation, will do it.

Reply:
I thought that our current shmem(transport) can be entirely mocked on the x86 host? 
Both sides of transport (server and client) can be invoked in sequential manner to make one complete message request response transaction. Bigger test functions can be written which will comprise multiple such function calls in sequence of both sides(server and client)

I am just thinking from a user perspective where i dont want to have qemu launch which will require me to first build it and also find the right repo and things which go with it.

Thoughts?

Reply:
> I thought that our current shmem(transport) can be entirely mocked on the x86 host?
> 
> Both sides of transport (server and client) can be invoked in sequential manner to make one complete message request response transaction. Bigger test functions can be written which will comprise multiple such function calls in sequence of both sides(server and client)
> 
> I am just thinking from a user perspective where i dont want to have qemu launch which will require me to first build it and also find the right repo and things which go with it.
> 
> Thoughts?

ofcourse, it can be mocked, as mentioned in options 2 or 3 [here](https://github.com/riscv-software-src/librpmi/pull/19#discussion_r1690923413). 

But.. see my concerns mentioned in the same comment. if we really want to we can copy qemu implementation for current and future service groups. I personally dont think its a good idea to push an test implementation which bloats up librpmi.